### DISCUSSION TITLE
CORS message and Conflicts & IBM Cloudant

### DISCUSSION THREAD

**Discussion by Vyeche** [vrtmrz/obsidian-livesync#121] [Q&A]
Hi, 

While using the live sync, I noticed that over time I build up conflicts or the false positive CORS message. I’ve seen that message before in .NET core and ASP.NET while using a C# API I built unrelated to IBM. My experience is that it’s usually a 500 error on the back end. Rebuilding and restarting doesn’t always fix the CORS issue while using multiple devices while batch sync is enabled. 

The issues come more frequently when I rename and move files in Obsidian.

I’ve logged into IBM Cloud and I noticed conflicts are building up, but I don’t know the API so I can’t fix it. 

Is there a way to view and fix the conflicts without going through the log? Is there an option to save the log? I checked out the menu for conflicts on the live sync plugin, but it always shows nothing. However, the log is showing issues. 

The do not delete empty folder is not being respected even though its set to false. Something is conflicting. 
https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/settings.md#do-not-delete-empty-folder

Thanks & kind regards,
Vye

**Comment by vrtmrz** [comment]
Hi! thank you for opening this discussion!

I wonder what messages are shown, but in conflicts, the remote conflicts are also local conflicts. So we can resolve them by `Pick a file to resolve conflict`. (If it shows nothing, we don't have to resolve them) 

> Is there an option to save the log?

Unfortunately not. But it might help us. I will implement it.

**Reply by Vyeche** [reply]
Yeah it showed nothing, when I rename files or move them it leaves folders behind or duplicates of the files. Especially when I use a different device with the same vault.

**Reply by vrtmrz** [reply]
Finally, I made it be reproduced!
The file tracking method has been refined. Still a bit hypersensitive, 
(sometimes detects own change as conflict) 
but it might be getting better than losing or making corrupted  files.
Could you please try 0.15.7?

**Reply by Vyeche** [reply]
Great, I will try it and get back to you.

**Reply by Vyeche** [reply]
I tested the file and folder clean up. I'm seeing the folder. But the file is gone. The setting for not deleting the folder in the plugin is not checked.

Why does the folder stay? Notice the test folder not deleted. 

![PXL_20220928_112347202.jpg](https://user-images.githubusercontent.com/972377/192766904-5e2442e4-4168-4d0a-880a-60806abf739b.jpg)


![PXL_20220928_112244060.jpg](https://user-images.githubusercontent.com/972377/192766724-1e9681a3-f6ed-4c6e-9baa-40484e7148de.jpg)

![PXL_20220928_112732283.jpg](https://user-images.githubusercontent.com/972377/192767621-2f75de4b-6b2b-4e9a-a26a-4e9e33d23fe1.jpg)

**Reply by vrtmrz** [reply]
Thank you for quick response! And sorry to disappoint you…
I will investigate it more.

**Reply by vrtmrz** [reply]
I finally discovered the mistake... I add a feature that filtering synshronisation target file on 0.14.0, but filter also folder as not a target. 
It would be fixed!

**Reply by Vyeche** [reply]
Cool, glad you found it ;)

**Reply by Vyeche** [reply]
@vrtmrz 
Thank you for the fixes. 

Here are some logs that I get all the time. Unfortunately, I don't have a quick method to resolve these without digging into each. As a result, the conflicts are not showing up on the plugin dashboard for me to fix them. 

The 500 error (CORS) is a false positive, probably with the corrupted miss match of a file (gut feeling).

![image](https://user-images.githubusercontent.com/972377/193170172-455a1b0c-b377-473b-9335-74489283d4eb.png)
![image](https://user-images.githubusercontent.com/972377/193169919-a047eb3e-ff28-4723-a37b-aa4d21112e91.png)
![image](https://user-images.githubusercontent.com/972377/193169963-d8434c56-36fe-4579-b0ef-cca186399c7d.png)
![image](https://user-images.githubusercontent.com/972377/193170344-9c1be8ef-1aee-4ee8-8a2b-011e0734f2b1.png)

**Reply by vrtmrz** [reply]
I have found several cases that LiveSync detects as CORS.
If there is an error, the error message will be detected as an illegal Cross-origin request.
~~Could you please click the red errored line and see the Response pane?~~ Probaly, there is more information. (This message will not reach Obsidian and LiveSync, because of security reasons.)
I will be able to investigate it more with the message, if the message is not that having exceeded your quota. I also have reached it several times...

Edited: Sorry, this cannot be shown. and now reproduced.

**Reply by vrtmrz** [reply]
![image](https://user-images.githubusercontent.com/45774780/193213505-d3da9b1b-e1c4-4c03-91d1-9516dc5d4149.png)
Is `429` shown anywhere? This means that we accessed the remote database too frequently. I didn't realised but as you said, perhaps we cannot use `Read chunks online` practically in the initial replication.
(or we have to delay the request file by file). 

I will improve the doc.

But, if you have some other status codes, please let me know. 

P.S. Once our synchronisation has been stable, we should be able to use `Read chunks online`. (That in my case)

**Reply by Vyeche** [reply]
214 requests came in and several show as 429.
Console is reporting 36 console CORS, setting console to blocked I see 18 in blocked requests. It doesn't seem to match the CORS.
![image](https://user-images.githubusercontent.com/972377/193296993-07c35241-5277-4ac5-b6d7-5190ffe0b81c.png)
![image](https://user-images.githubusercontent.com/972377/193296766-606a857f-273a-4b18-b7de-52f4b9e6b606.png)
![image](https://user-images.githubusercontent.com/972377/193297222-fac9b929-c8c4-4cfa-90f1-38e5d0dc07ef.png)
![image](https://user-images.githubusercontent.com/972377/193295107-94874400-a4be-49c7-b3c4-457592e55d39.png)
![image](https://user-images.githubusercontent.com/972377/193295414-20ec96fe-96d5-49bd-8e93-c034068dd7a9.png)

**Reply by vrtmrz** [reply]
Thank you for your investigation! I am relieved that it was exactly as I thought.
It is not good for Cloudant. It also will be a performance issue in general cases. 
It has been fixed in 0.15.9.

**Reply by Vyeche** [reply]
That’s great, i will try it out.
